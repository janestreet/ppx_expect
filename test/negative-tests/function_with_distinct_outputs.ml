module Expect_test_config = struct
  include Expect_test_config

  let upon_unreleasable_issue = `Warning_for_collector_testing
end

module Expectation = Ppx_expect_runtime.For_external.Expectation
[@@alert "-ppx_expect_runtime"]

let%expect_test _ =
  let f output =
    print_string output;
    [%expect {| hello world |}]
  in
  f "foo";
  f "bar"
;;

let%expect_test "same expectation with multiple inconsistent outputs" =
  let test output =
    print_endline output;
    [%expectation {||}];
    Expectation.commit ()
  in
  test "foo";
  test "bar"
;;
