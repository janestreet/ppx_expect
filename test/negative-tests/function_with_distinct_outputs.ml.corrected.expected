module Expect_test_config = struct
  include Expect_test_config

  let upon_unreleasable_issue = `Warning_for_collector_testing
end

module Expectation = Ppx_expect_runtime.For_external.Expectation
[@@alert "-ppx_expect_runtime"]

let%expect_test _ =
  let f output =
    print_string output;
    [%expect
      {|
      (* expect_test: Test ran multiple times with different test outputs *)
      ============================ Output 1 / 2 ============================
      foo
      ============================ Output 2 / 2 ============================
      bar
      |}]
  in
  f "foo";
  f "bar"
;;

let%expect_test "same expectation with multiple inconsistent outputs" =
  let test output =
    print_endline output;
    [%expectation
      {|
      (* expect_test: Test ran multiple times with different test outputs *)
      ============================ Output 1 / 2 ============================
      foo

      ============================ Output 2 / 2 ============================
      bar
      |}];
    Expectation.commit ()
  in
  test "foo";
  test "bar"
;;
